### PR TITLE
feat: implement initial version of p2p relay server

### PR DESCRIPTION
**Pull request by fbozic**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: Release
+on:
+  push:
+    branches:
+      - master
+      - feat/initial-implementation
```
Comment (MatejVukosav) [inline .github/workflows/release.yml:6]: do you still need this feat/

**Review by MatejVukosav — APPROVED** [review]

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    ping: ping::Behaviour,
+    identify: identify::Behaviour,
+}
+
+#[derive(Debug, Parser)]
+#[clap(name = "calimero relay")]
+struct Opt {
+    /// Fixed value to generate deterministic peer id
+    #[clap(long)]
+    secret_key_seed: u8,
```
Comment (miraclx) [inline src/main.rs:24]: just collect `--peer-id` instead

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        handle_swarm_event(&mut swarm, event.expect("Swarm stream to be infinite.")).await;
+    }
+}
+
+fn generate_ed25519(secret_key_seed: u8) -> identity::Keypair {
+    let mut bytes = [0u8; 32];
+    bytes[0] = secret_key_seed;
+
+    identity::Keypair::ed25519_from_bytes(bytes).expect("only errors on wrong length")
+}
```
Comment (miraclx) [inline src/main.rs:84]: so we can dump this

**Review by miraclx — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Build for Intel Linux
+        run: cargo build --release --target=x86_64-unknown-linux-gnu
+        if: steps.check_release.outputs.exists == 'false'
+
+      - name: Build for ARM Linux
+        run: cross build --release --target=armv7-unknown-linux-gnueabihf
+        if: steps.check_release.outputs.exists == 'false'
+
+      - name: Create artifacts directory
+        run: |
```
Comment (miraclx) [inline .github/workflows/release.yml:53]: I think you mean to skip this too?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        run: rustup toolchain install stable --profile minimal
+
+      - name: Setup rust cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Get version
+        id: get_version
+        run: echo "version=$(cargo read-manifest | jq -r '.version')" >> $GITHUB_OUTPUT
+
+      - name: Check if release exists
```
Comment (miraclx) [inline .github/workflows/release.yml:26]: why not break this up into an initial job that makes the build & release one conditional?

https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs

**Pull request merged**